google/cloud/aiplatform/model_monitoring/objective.py

```diff
@@ -0,0 +1,296 @@
+        """
+        super().__init__(
+            data_source,
+            skew_thresholds,
+            target_field,
+            attribute_skew_thresholds,
+            data_format,
+        )
+
+        training_dataset = (
+            gca_model_monitoring.ModelMonitoringObjectiveConfig.TrainingDataset(
+                target_field=target_field
+            )
+        )
+        if data_source.startswith("bq:/"):
+            training_dataset.bigquery_source = gca_io.BigQuerySource(
+                input_uri=data_source
+            )
+        elif data_source.startswith("gs:/"):
+            training_dataset.gcs_source = gca_io.GcsSource(uris=[data_source])
+            if data_format is not None and data_format not in [
+                "tf-record",
+                "csv",
+                "jsonl",
+            ]:
+                raise ValueError(
+                    "Unsupported value. `data_format` must be one of 'tf-record', 'csv', or 'jsonl'"
+                )
+            training_dataset.data_format = data_format
+        else:
+            training_dataset.dataset = data_source
+        self.training_dataset = training_dataset
+
+
+class EndpointDriftDetectionConfig(_DriftDetectionConfig):
+    """A class that configures prediction drift detection for models deployed to an endpoint.
+
+    Prediction drift occurs when feature data distribution changes noticeably
+    over time, and should be set when the original training data is unavailable.
+    If original training data is available, EndpointSkewDetectionConfig should
+    be set instead.
+    """
+
+    def __init__(
+        self,
+        drift_thresholds: Optional[Dict[str, float]] = {},
+        attribute_drift_thresholds: Optional[Dict[str, float]] = {},
+    ):
+        """Initializer for EndpointDriftDetectionConfig
+
+        Args:
+            drift_thresholds (Dict[str, float]):
+                Optional. Key is the feature name and value is the
+                threshold. If a feature needs to be monitored
+                for drift, a value threshold must be configured
+                for that feature. The threshold here is against
+                feature distribution distance between different
+                time windws.
+
+            attribute_drift_thresholds (Dict[str, float]):
+                Optional. Key is the feature name and value is the
+                threshold. The threshold here is against
+                attribution score distance between different
+                time windows.
+
+        Returns:
+            An instance of EndpointDriftDetectionConfig.
+        """
+        super().__init__(drift_thresholds, attribute_drift_thresholds)
+
+
+class EndpointExplanationConfig(_ExplanationConfig):
+    """A class that enables Vertex Explainable AI.
+
+    Only applicable if the model has explanation_spec populated.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.enable_feature_attributes = True
```

Comment:
This was a design choice made by Morgan, and I just went along with the approved design doc. My guess is that it slightly reduces the amount of code that developers have to write, since the XAI config defaults to `False`. Therefore instantiating the class implies that the users want to turn the config on. I added a line in the docs to indicate that the default behavior is to disable XAI.